File: ProjectManagement/Models/Project.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace ProjectManagement.Models;
public class Project
{
    public Guid Id { get; set; } = Guid.Empty;
    [Required(AllowEmptyStrings = false ,ErrorMessage = "Project name is required")]
    public string Name { get; set; } = string.Empty;
    public DateTime CreatedDate { get; set; } = DateTime.Now;
    public DateTimeOffset Deadline { get; set; }
    public List<ProjectTask>? Tasks { get; set; } = [];
    public int? StatusId {get; set; } 
    [JsonIgnore]
    public Status? Status { get; set; }
}

File: web-front/src/models/project.ts
import { Task } from './task';
import { Status } from './status';

export class Project {
    id: string;
    name: string;
    createdDate: Date = new Date();
    deadline: string;
    tasks?:  Task[] = [];
    statusId?: number;
    status?: Status;

    constructor(id: string,  name: string,  createdDate: Date, statusId: number, deadline: string, tasks?: Task[], status?: Status) {
        this.id = id;
        this.name = name;
        this.createdDate = createdDate;
        this.deadline = deadline;
        this.tasks = tasks;
        this.statusId = statusId;
        this.status = status;
    }
}